src/NPRKTrees.jl

```diff
@@ -0,0 +1,173 @@
+		# if order is 1, generate trivial trees with root color 0<=i<=num_partitions-1
+		for i in 0:num_partitions-1
+			trivialtree = rootedtree([1],[i])
+			push!(toReturn,trivialtree)
+		end
+	else
+		# else order > 1
+		# Generate all possible color sequences with colorings 0,1,...,num_partitions-1
+		# 	of length order
+		colors = reverse.(Iterators.product(fill(0:num_partitions-1,order)...))[:]
+
+		# Loop over all uncolored rooted trees and append the above color sequences
+		# 	to define colored rooted trees
+		# Note that this will generate duplicate trees that are isomorphic
+		for tree in RootedTreeIterator(order)
+			for c in colors
+				colorseq = collect(c)
+				push!(toReturn, rootedtree(tree.level_sequence,colorseq))
+			end
+		end
+	end
+	# remove isomorphic trees
+	toReturn = unique(toReturn,dims=1)
+	return toReturn
+end;
+
+# Generates all non-isomorphic NPRK trees with num_partitions (i.e., edge colors) of a given order
+#   equivalent to all ARK trees with num_partition colors of that order, with root node coloring fixed = 0
+# 	returned as a Vector of ColoredRootedTree's
+function NPRKTreeIterator(num_partitions::Int64, order::Int64)::Vector{<:ColoredRootedTree}
+	toReturn = ColoredRootedTree[]
+	
+	if num_partitions < 1
+		throw(DomainError(num_partitions,"num_partitions argument must be integer >= 1"))
+	elseif order < 1
+		throw(DomainError(order,"order argument must be integer >= 1"))
+	elseif order == 1
+		# if order is 1, generate trivial tree with fixed root color 0
+		trivialtree = rootedtree([1],[0])
+		push!(toReturn,trivialtree)
+	else
+		# else order > 1:
+		# Generate all possible color sequences with colorings 0,1,...,partitions-1
+		# 	of length order-1 (minus one because the root coloring is fixed to 0)
+		colors = reverse.(Iterators.product(fill(0:num_partitions-1,order-1)...))[:]
+	
+		# Loop over all uncolored rooted trees and append the above color sequences
+		# 	to define colored rooted trees with fixed root color 0
+		# Note that this will generate duplicate trees that are isomorphic
+		for tree in RootedTreeIterator(order)
+			for c in colors
+				colorseq = collect(c)
+				pushfirst!(colorseq,0) # push 0 to start of color sequence
+				push!(toReturn, rootedtree(tree.level_sequence,colorseq))
+			end
+		end
+	end
+	# remove isomorphic trees
+	toReturn = unique(toReturn,dims=1)
+
+	return toReturn
+end;
+
+# Determines whether a ColoredRootedTree is multicolored:
+# 	argument root_color_fixed = true for NPRK trees, i.e., checks if edges of the NPRK tree are multicolored
+# 	argument root_color_fixed = false for ARK trees, i.e., checks if nodes of the NPRK tree are multicolored
+# 		corresponds to additive coupling conditions
+function isMultiColored(colored_tree::ColoredRootedTree, root_color_fixed::Bool)::Bool
+	if root_color_fixed
+		colorseq = copy(colored_tree.color_sequence)
+		popfirst!(colorseq)
+		return ~allequal(colorseq)
+	else
+		return ~allequal(colored_tree.color_sequence)
```

Comment:
```suggestion
		return !allequal(colorseq)
	else
		return !allequal(colored_tree.color_sequence)
```

`~` is bitwise not, while `!` is the general not operation that is more common in Julia code.